pkg/influx/parser_test.go

```diff
@@ -0,0 +1,139 @@
+				{
+					Labels:  []client.LabelAdapter{{Name: "__name__", Value: "measurement_f1"}, {Name: "t1", Value: "v1"}, {Name: "t2", Value: "v2"}, {Name: "t3", Value: "v3"}},
+					Samples: []client.Sample{{Value: 36, TimestampMs: 1465839830100}},
+				},
+			},
+		},
+		{
+			name: "parse multiple fields",
+			url:  "/",
+			data: "measurement,t1=v1 f1=3.0,f2=365,f3=0 1465839830100400200",
+			expectedResult: []client.TimeSeries{
+				{
+					Labels:  []client.LabelAdapter{{Name: "__name__", Value: "measurement_f1"}, {Name: "t1", Value: "v1"}},
+					Samples: []client.Sample{{Value: 3, TimestampMs: 1465839830100}},
+				},
+				{
+					Labels:  []client.LabelAdapter{{Name: "__name__", Value: "measurement_f2"}, {Name: "t1", Value: "v1"}},
+					Samples: []client.Sample{{Value: 365, TimestampMs: 1465839830100}},
+				},
+				{
+					Labels:  []client.LabelAdapter{{Name: "__name__", Value: "measurement_f3"}, {Name: "t1", Value: "v1"}},
+					Samples: []client.Sample{{Value: 0, TimestampMs: 1465839830100}},
+				},
+			},
+		},
+		{
+			name: "parse invalid char conversion",
+			url:  "/",
+			data: "*measurement,#t1?=v1 f1=0 1465839830100400200",
+			expectedResult: []client.TimeSeries{
+				{
+					Labels:  []client.LabelAdapter{{Name: "__name__", Value: "_measurement_f1"}, {Name: "_t1_", Value: "v1"}},
+					Samples: []client.Sample{{Value: 0, TimestampMs: 1465839830100}},
+				},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			req := httptest.NewRequest("POST", tt.url, bytes.NewReader([]byte(tt.data)))
+
+			timeSeries, err := parseInfluxLineReader(context.Background(), req, maxSize)
+			require.NoError(t, err)
+			for i := 0; i < len(timeSeries); i++ {
+				assert.Equal(t, timeSeries[i].String(), tt.expectedResult[i].String())
+			}
+		})
+	}
+}
+
+func TestInvalidInput(t *testing.T) {
+	tests := []struct {
+		name string
+		url  string
+		data string
+	}{
+		{
+			name: "parse invalid precision",
+			url:  "/write?precision=ss", // precision must be of type "ns", "us", "ms", "s"
+			data: "measurement,t1=v1 f1=2 1465839830100400200",
+		},
+		{
+			name: "parse invalid field input",
+			url:  "/write",
+			data: "measurement,t1=v1 f1= 1465839830100400200", // field value is missing
+		},
+		{
+			name: "parse invalid tags",
+			url:  "/write",
+			data: "measurement,t1=v1,t2 f1=2 1465839830100400200", // field value is missing
+		},
+		{
+			name: "parse field value invalid quotes",
+			url:  "/write",
+			data: "measurement,t1=v1 f1=v1 1465839830100400200", // string type field values require double quotes
+		},
+		{
+			name: "parse missing field",
+			url:  "/write",
+			data: "measurement,t1=v1 1465839830100400200", // missing field
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			req := httptest.NewRequest("POST", tt.url, bytes.NewReader([]byte(tt.data)))
+
+			_, err := parseInfluxLineReader(context.Background(), req, maxSize)
+			require.Error(t, err)
```

Comment:
do we return different error types for each issue? if so then we want to confirm that the correct error is returned for each case.  Even if they all share the same error type we still probably want to test that we get a specific error type (errors.As) so we know that it will get processed as a client, not server error

Reply:
Yes, just added that

Reply:
sorry for the delay on this.  Is it not possible to check errors by type instead of message?  It's generally not a good idea to check by message contents because that can change and cause a breakage in the test.  However if all the error types are the same, then this is the way to go.  I'll mark LGTM to unblock, but if it's possible to check by type please do that.

Reply:
I will be filing an issue to improve the errors so that error types are returned instead of fmt.errors